src/compiler/checker.ts

```diff
@@ -10449,7 +10449,10 @@ namespace ts {
                     case SyntaxKind.FunctionExpression:
                     case SyntaxKind.FunctionDeclaration:
                     case SyntaxKind.ArrowFunction:
-                    case SyntaxKind.ObjectLiteralExpression: return false;
+                    case SyntaxKind.ClassExpression:
+                    case SyntaxKind.ClassDeclaration:
+                    case SyntaxKind.GetAccessor:
+                    case SyntaxKind.SetAccessor: return false;

```

Comment:
I'd use `isFunctionLike` and `isClassLike` if it exists. And I think given that, you can basically just get rid of the `switch`.